bridgerelayer/substrate/client.go

```diff
@@ -0,0 +1,131 @@
+package substrate
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	gsrpc "github.com/Snowfork/go-substrate-rpc-client"
+	"github.com/Snowfork/go-substrate-rpc-client/config"
+	"github.com/Snowfork/go-substrate-rpc-client/signature"
+	"github.com/Snowfork/go-substrate-rpc-client/types"
+
+	etypes "github.com/snowfork/polkadot-ethereum/bridgerelayer/types"
+)
+
+// Client struct
+type Client struct {
+	api      *gsrpc.SubstrateAPI
+	metadata *types.Metadata
+}
+
+// NewClient foo
+func NewClient() (*Client, error) {
+
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		panic(err)
+	}
+
+	client := Client{
+		api, metadata,
+	}
+
+	return &client, nil
+}
+
+// SubmitExtrinsic submits a packet
+func (client *Client) SubmitExtrinsic(appID [32]byte, packet etypes.PacketV2) (bool, error) {
+
+	from, err := signature.KeyringPairFromSecret("//Alice", "")
+	if err != nil {
+		return false, err
+	}
+
+	meta, err := client.api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		return false, err
+	}
+
+	appid := types.NewBytes32(appID)
+
+	payload, err := types.EncodeToBytes(packet)
+	if err != nil {
+		return false, err
+	}
+
+	// used for debugging
+	_err := ioutil.WriteFile("/tmp/packet.scale", payload, 0644)
```

Comment:
```suggestion
	err := ioutil.WriteFile("/tmp/packet.scale", payload, 0644)
```

Reply:
Can we add a debug flag to the cli to trigger debug mode?

Reply:
I think here we'll need `err = ioutil.WriteFile("/tmp/packet.scale", payload, 0644)` as err has already been declared.

Reply:
I've removed the debug code for now.